### ISSUE TITLE
ContentView support?

### ISSUE THREAD

**Issue by VagyokC4** [rid00z/FreshMvvm#208]
Hi Michael,

Awesome library.  Any plans to support ContentView's  I'd like to be able to resolve ContentView's on demand in my PageModel and get all the MVVM magic FreshMVVM brings to the table.

My end goal is to be able to, based on the data coming in, resolve various ContentViews and build out the page.

**Comment by EmilAlipiev** [comment]
how do you want to resolve contentview? contentview is a part of contentpage so, it couples with pagemodel (viewmodel) of the contentpage itself. that means if you use contentview in  APage, your viewmodel is  APagemodel, if you use in  BPage, your viewmodel for same contentview is  BPageModel. You can also use a BasePageModel inherits on APageModel and BPageModel. In this case your ContentView can have either BasePageModel or APageModel/BPageModel.. what else can it be done here?

**Comment by VagyokC4** [comment]
@EmilAlipiev Thank-you for your response.

Basically I want to have a stack-layout with various content-view as it's children on APage with their content-view-content databound to content-view properties on APageModel.  Based on the data passed into APageModel, I want to resolve the various content-view-models that plug into the content-view properties.

I would expect each content-page and content-view to have their own page/view-model.  I would expect APageModel to resolve AContentViewModel resulting in the content-view, along with ability to get to the content-view-model being resolved, as well as the ability to pass data into the content-view-model when resolving.

Basically the content-view-model should have the same life cycle events and functionality that the current resolve<T>() has with content-page, with the exception that it returns a content-view instead of a content-page.

**Comment by EmilAlipiev** [comment]
Contentview always respects to its contantentpage's bindingcontext. You can see that checking in code behind of contentview using this.bindingcontext. It cant have a separate bindingcontext than its parent.

**Comment by VagyokC4** [comment]
@EmilAlipiev Ok.  So ContentView-A and ContentView-B would both use ContentPage-A's view model for binding.  Can we Resolve multiple content-view's and plug them in?

**Comment by EmilAlipiev** [comment]
Sorry your question is not clear. maybe you should give an example of usage that you tried but it failed. And why would you need something like that? what is the benefit of it? contentview is containter not a page. it is no different than a stacklayout or grid. you are asking to have viewmodel for a container. I havent programmed Wpf long time  but i dont remember such thing existed in either of the Xaml- mvvm based technologies.

**Comment by schovan** [comment]
Hi,
if I uderstand it correctly, @VagyokC4 wants to support something similar like this in Caliburn.Micro framework:
```
<ContentView cal:View.Model="{Binding ViewModel}" />
```
You can switch ConentView's content base on ViewModel property. You assign some View Model (FreshBasePageModel in this case) to the property and appropriate View (Page in this case) is automatically resolved and displayed.
Is it possile to do this in FreshMvvm framework (and FreshMvvm.Maui)?
If not, can you consider adding it to the framework?
